pkg/apiserver/proxy_test.go

```diff
@@ -36,122 +38,219 @@ import (
 	"k8s.io/kubernetes/pkg/api/rest"
 )
 
-func calculateContentLength(reqBody string, transferEncodings []string) int64 {
-	if len(transferEncodings) == 0 || (len(transferEncodings) == 1 && transferEncodings[0] == "identity") {
-		return int64(len(reqBody))
+func TestProxyRequestContentLengthAndTransferEncoding(t *testing.T) {
+	chunk := func(data []byte) []byte {
+		out := &bytes.Buffer{}
+		chunker := httputil.NewChunkedWriter(out)
+		for _, b := range data {
+			if _, err := chunker.Write([]byte{b}); err != nil {
+				panic(err)
+			}
+		}
+		chunker.Close()
+		out.Write([]byte("\r\n"))
```

Comment:
Hmm...I see your code is aligned with the practice in transfer.go, so it should be correct.

Out of curiosity, `out.Write([]byte("\r\n"))` adds the final CRLF, but who writes the "last-chunk" specified in rfc2616?